### PR TITLE
Fix NoneType AttributeError in firefox extension plugin

### PR DESCRIPTION
**Pull request by M1ra1B0T**
This PR fixes a NoneType AttributeError in the firefox extension plugin.
When parsing the firefox extension information from the json file, we use the dictionary get method to access the relevant key value pairs. However, I ran into an uncatched NoneType AttributeError if the original key value is None, but we expect a nested dictionary, e.g. for the "name" field.

Additionally, the error message was slightly adapted, so instead of logging the whole user object, it now only prints the username as it should.

### CODE REVIEW

**Comment by Schamper** [comment]
Do you have some test data or could you add some data to the unit test for this? Maybe I'm missing something but on first glance the previous code should be functionality identical?

```python
In [1]: extension = {}

In [2]: repr(extension.get("defaultLocale", {}).get("name"))
Out[2]: 'None'

In [3]: extension = {"defaultLocale": {}}

In [4]: repr(extension.get("defaultLocale", {}).get("name"))
Out[4]: 'None'
```

**Comment by M1ra1B0T** [comment]
@Schamper The error occurs if the field `defaultLocale` is actually `None`.

```
>>> extension = {"defaultLocale": None}
>>> repr(extension.get("defaultLocale", {}).get("name"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

File: dissect/target/plugins/apps/browser/firefox.py [hunk 1]
```diff
@@ -363,25 +363,33 @@ def extensions(self) -> Iterator[BrowserExtensionRecord]:
                         ts_update=extension.get("updateDate", 0) // 1000,
                         browser="firefox",
                         id=extension.get("id"),
-                        name=extension.get("defaultLocale", {}).get("name"),
+                        name=None
+                        if extension.get("defaultLocale") is None
+                        else extension.get("defaultLocale", {}).get("name"),
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/firefox.py:368]:
```suggestion
                        name=(extension.get("defaultLocale", {}) or {}).get("name"),
```

File: dissect/target/plugins/apps/browser/firefox.py [hunk 2]
```diff
@@ -363,25 +363,33 @@ def extensions(self) -> Iterator[BrowserExtensionRecord]:
...
                         short_name=None,
                         default_title=None,
-                        description=extension.get("defaultLocale", {}).get("description"),
+                        description=None
+                        if extension.get("defaultLocale") is None
+                        else extension.get("defaultLocale", {}).get("description"),
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/firefox.py:373]:
```suggestion
                        description=(extension.get("defaultLocale", {}) or {}).get("description"),
```

File: dissect/target/plugins/apps/browser/firefox.py [hunk 3]
```diff
@@ -363,25 +363,33 @@ def extensions(self) -> Iterator[BrowserExtensionRecord]:
...
                         version=extension.get("version"),
                         ext_path=extension.get("path"),
                         from_webstore=None,
-                        permissions=extension.get("userPermissions", {}).get("permissions"),
+                        permissions=None
+                        if extension.get("userPermissions") is None
+                        else extension.get("userPermissions", {}).get("permissions"),
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/firefox.py:379]:
```suggestion
                        permissions=(extension.get("userPermissions", {}) or {}).get("permissions"),
```

File: dissect/target/plugins/apps/browser/firefox.py [hunk 4]
```diff
@@ -363,25 +363,33 @@ def extensions(self) -> Iterator[BrowserExtensionRecord]:
...
                         manifest_version=extension.get("manifestVersion"),
                         source_uri=extension.get("sourceURI"),
-                        optional_permissions=extension.get("optionalPermissions", {}).get("permissions"),
+                        optional_permissions=None
+                        if extension.get("optionalPermissions") is None
+                        else extension.get("optionalPermissions", {}).get("permissions"),
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/firefox.py:384]:
```suggestion
                        optional_permissions=(extension.get("optionalPermissions", {}) or {}).get("permissions"),
```

**Review by Schamper — CHANGES_REQUESTED** [review]
Suggested something a bit more concise.

**Review by Schamper — APPROVED** [review]

**Comment by Schamper** [comment]
Thanks! Was actually running into this issue too while working on another feature.

**Pull request merged**